go/enclave/crypto/da_enc_service.go

```diff
@@ -58,26 +67,38 @@ func createCypher(sharedSecretService *SharedSecretService) (*cipher.AEAD, error
 	if err != nil {
 		return nil, fmt.Errorf("could not initialise GCM cipher for enclave DA key. cause %w", err)
 	}
-	return &cipher, nil
+	return cipher, nil
 }
 
 func (t *DAEncryptionService) Encrypt(blob []byte) ([]byte, error) {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+
 	if t.cipher == nil {
 		return nil, errors.New("not initialised")
 	}
 
 	nonce, err := generateSecureEntropy(GCMNonceLength)
 	if err != nil {
 		t.logger.Error("could not generate nonce to encrypt transactions.", log.ErrKey, err)
-		return nil, err
+		return nil, fmt.Errorf("nonce generation failed: %w", err)
 	}
 
-	ciphertext := (*t.cipher).Seal(nil, nonce, blob, nil)
-	// We prepend the nonce to the ciphertext, so that it can be retrieved when decrypting.
-	return append(nonce, ciphertext...), nil //nolint:makezero
+	result := make([]byte, GCMNonceLength+len(blob)+t.cipher.Overhead())
```

Comment:
is this just a semantic change over the return?

Reply:
it was suggested by the AI. More memory efficient.